luigi/contrib/ecs.py

```diff
@@ -163,6 +166,73 @@ def command(self):
         """
         pass
 
+    @property
+    def combined_overrides(self):
+        """
+        Return single dict combining any provided `overrides` parameters.
+
+        This is used to allow custom `overrides` parameters to be specified in
+        `self.run_task_kwargs` while ensuring that the values specified in
+        `self.command` are honored in `containerOverrides`.
+        """
+        overrides = copy.deepcopy(self.run_task_kwargs.get('overrides', {}))
+        if self.command:
+            if 'containerOverrides' in overrides:
+                new_commands = []
+                for command in self.command:
+                    for container_override in overrides['containerOverrides']:
+                        if container_override['name'] == command['name']:
```

Comment:
this is some deep nesting and is a little hard to follow the logic. Is there a way this could be cleaned up or visually simplified?

Reply:
Sure, I will take a look at it this afternoon.

Reply:
Just pushed a new commit that hopefully makes it clearer, let me know what you think.